### PR TITLE
Add OIDC config to add extra parameters to the authorize URL

### PR DESCRIPTION
**Pull request by MatMaul**
Add an OIDC config to specify extra parameters for the authorization grant URL.

It can be useful to pass an ACR value for example.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docs/usage/configuration/config_documentation.md [hunk 1]
```diff
@@ -3473,6 +3476,8 @@ oidc_providers:
     token_endpoint: "https://accounts.example.com/oauth2/token"
     userinfo_endpoint: "https://accounts.example.com/userinfo"
     jwks_uri: "https://accounts.example.com/.well-known/jwks.json"
+    extra_grant_values:
```
Comment (reivilibre) [inline docs/usage/configuration/config_documentation.md:3479]: The answer might be no, but would it make sense to rename this to `extra_authorization_parameters`?

I think this is just extra URL-params to the authorisation endpoint right?

Comment (MatMaul) [reply docs/usage/configuration/config_documentation.md:3479]: After reading a bit more about OIDC and English semantics you can get a grant after calling for authorization, and the endpoint we passed those params to is indeed `/authorize`.

So I agree with you.

@sandhose for a quick sanity check from the expert by I am going to rename it for now.

Comment (MatMaul) [reply docs/usage/configuration/config_documentation.md:3479]: I am thinking to use `extra_authorize_parameters` to match the endpoint name however.

Comment (sandhose) [reply docs/usage/configuration/config_documentation.md]: For what it's worth, I named this option `additional_authorization_parameters` in MAS. That being said, I don't really have a strong opinion on this

Comment (MatMaul) [reply docs/usage/configuration/config_documentation.md]: I am fine with that and it makes sense to unify to ease the migration, on it thanks !

Comment (MatMaul) [reply docs/usage/configuration/config_documentation.md]: And done.

**Review by reivilibre — APPROVED** [review]
Thanks!

**Comment by reivilibre** [comment]
but the lints are not happy :(

**Comment by MatMaul** [comment]
Arf sorry I should have checked :( let's see this run, it passes locally for the lint at least.

**Pull request merged**